core/src/main/scala/swave/core/StreamOps.scala

```diff
@@ -312,6 +312,46 @@ trait StreamOps[A] extends Any { self ⇒
   final def slice(startIndex: Long, length: Long): Repr[A] =
     via(Pipe[A] drop startIndex take length named "slice")
 
+  final def sliding(windowSize: Int, emitSingleEmpty: Boolean = false): Repr[immutable.Seq[A]] =
+    slidingTo[immutable.Seq](windowSize, emitSingleEmpty)
+
+  final def slidingTo[M[+ _]](windowSize: Int, emitSingleEmpty: Boolean = false)(
+      implicit cbf: CanBuildFrom[M[A], A, M[A]]): Repr[M[A]] = {
+
+    val builder = cbf.apply()
+
+    prefixAndTail(windowSize).map {
```

Comment:
There is a special case to be handled - if the stream is shorter than the `windowSize`, it should only produce one sequence (one unfull window). If there are more elements than `windowSize`, then we need a "warmup" phase, filling the buffer with the first window elements before emitting the first window (a single scan would also emit halt full windows during the warmup). I'd like to remove the prefixAndTail, however, I don't quite know how to do it with a single scan. Doesn't a scan always produce n elements for a stream with n elements? Maybe I'm missing something...